### PR TITLE
Migration of test_gluster_volume_status_xml_dump

### PR DESCRIPTION
**Pull request by nishith-vihar**
[test_gluster_volume_status_xml_dump.py](https://github.com/gluster/glusto-tests/blob/master/tests/functional/glusterd/test_gluster_volume_status_xml_dump.py) test case is migrated to redant.

Updates: #292
Fixes: #335 
Signed-off-by: Nishith Vihar Sakinala nsakinal@redhat.com

### CODE REVIEW

File: tests/functional/glusterd/test_gluster_volume_status_xml_dump.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# nonDisruptive;dist-arb
+from time import sleep
+from tests.abstract_test import AbstractTest
+
+
+class GetVolumeStatusXmlDump(AbstractTest):
+
+    def run_test(self, redant):
+        """
+        Setps:
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_gluster_volume_status_xml_dump.py:32]: nit: Steps

File: tests/functional/glusterd/test_gluster_volume_status_xml_dump.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License along
+  with this program; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+
+Description:
+    Test Default volume behavior and quorum options
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_gluster_volume_status_xml_dump.py:20]: I don't think we are checking any quorum options, this can be updated.

**Review by nik-redhat — CHANGES_REQUESTED** [review]
Also, the test requires testing it with 2 or more volumes, so create and start another volume in the test itself, then stop the initially created volume and check the status dump. Delete the intermediate volume in the end.

File: tests/functional/glusterd/test_gluster_volume_status_xml_dump.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            redant.logger.error("Failed to get volume status on "
+                                f"{self.server_list[0]}")
+
+        for _ in range(4):
+            sleep(2)
+            out1 = redant.get_volume_status(self.mnode)
+            if out1 is None:
+                redant.logger.error("Failed to get volume status on "
+                                    f"{self.server_list[0]}")
+            self.assertEqual(out1, out)
```
Comment (srijan-sivakumar) [inline tests/functional/glusterd/test_gluster_volume_status_xml_dump.py:50]: I don't remember seeing assertEqual function in out codebase..can you point out as to where we have defined this ?

File: tests/functional/glusterd/test_gluster_volume_status_xml_dump.py (hunk 1 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_gluster_volume_status_xml_dump.py:32]: Ack

File: tests/functional/glusterd/test_gluster_volume_status_xml_dump.py (hunk 3 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_gluster_volume_status_xml_dump.py:50]: Thats a mistake. it was in glusto. Will remove it

File: tests/functional/glusterd/test_gluster_volume_status_xml_dump.py (hunk 2 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_gluster_volume_status_xml_dump.py:20]: Ack

**Comment by nishith-vihar** [comment]
> Also, the test requires testing it with 2 or more volumes, so create and start another volume in the test itself, then stop the initially created volume and check the status dump. Delete the intermediate volume in the end.

any specific volume type to be created or any vol type can be created?

**Comment by nik-redhat** [comment]
> > Also, the test requires testing it with 2 or more volumes, so create and start another volume in the test itself, then stop the initially created volume and check the status dump. Delete the intermediate volume in the end.
> 
> any specific volume type to be created or any vol type can be created?

Anything, for simplicity, let's keep it a distributed volume with 2/3 bricks.

File: tests/functional/glusterd/test_gluster_volume_status_xml_dump.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        Setps:
+        1. stop one of the volume
+            (i.e) gluster volume stop <vol-name>
+        2. Get the status of the volumes with --xml dump
+            XML dump should be consistent
+        """
+        redant.volume_stop(self.vol_name, self.server_list[0], force=True)
+        out = redant.get_volume_status(self.mnode)
+        if out is None:
+            redant.logger.error("Failed to get volume status on "
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_gluster_volume_status_xml_dump.py:41]: It should raise an exception instead.

File: tests/functional/glusterd/test_gluster_volume_status_xml_dump.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                                f"{self.server_list[0]}")
+
+        for _ in range(4):
+            sleep(2)
+            out1 = redant.get_volume_status(self.mnode)
+            if out1 is None:
+                redant.logger.error("Failed to get volume status on "
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_gluster_volume_status_xml_dump.py:48]: Same as above.

File: tests/functional/glusterd/test_gluster_volume_status_xml_dump.py (hunk 4 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_gluster_volume_status_xml_dump.py:41]: Ack

File: tests/functional/glusterd/test_gluster_volume_status_xml_dump.py (hunk 5 above)
Comment (nishith-vihar) [reply tests/functional/glusterd/test_gluster_volume_status_xml_dump.py:48]: Ack

File: tests/functional/glusterd/test_gluster_volume_status_xml_dump.py (hunk 1 above)
Comment (srijan-sivakumar) [reply tests/functional/glusterd/test_gluster_volume_status_xml_dump.py:32]: nit is still active :p Let's not resolve it before either fixing it or adding a relevant response.

**Review by srijan-sivakumar — CHANGES_REQUESTED** [review]

**Comment by srijan-sivakumar** [comment]
@nishith-vihar for best practice, don't resolve the conversations until and unless either,
1. You've changed the code after discussion.
2. Provided some explanation as to why the said suggestion is not taken up.

**Comment by nishith-vihar** [comment]
Screenshot from 2021-05-19 13-43-04

File: tests/functional/glusterd/test_gluster_volume_status_xml_dump.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            sleep(2)
+            out1 = redant.get_volume_status(node=self.server_list[0])
+            if out1 is None:
+                raise Exception("Failed to get volume status on "
+                                f"{self.server_list[0]}")
+            if out1 != out:
+                raise Exception
+
+        redant.volume_stop(volume_name1, self.server_list[0])
+        redant.volume_delete(volume_name1, self.server_list[0])
```
Comment (nik-redhat) [inline tests/functional/glusterd/test_gluster_volume_status_xml_dump.py:62]: In the future, the addition of the `volume_cleanup` function for these kind of tests would be helpful.

**Review by nik-redhat — APPROVED** [review]

File: tests/functional/glusterd/test_gluster_volume_status_xml_dump.py (hunk 6 above)
Comment (srijan-sivakumar) [reply tests/functional/glusterd/test_gluster_volume_status_xml_dump.py:62]: Ack.

**Review by srijan-sivakumar — APPROVED** [review]
LGTM

**Pull request merged**